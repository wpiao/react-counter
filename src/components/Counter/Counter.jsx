import React, { Component } from "react";

class Counter extends Component {
  constructor(props) {
    super(props);

    this.state = {
      counter: 0,
    };
  }

  increment = () => {
    this.setState({ counter: this.state.counter + 1 });
  };

  decrement = () => {
    this.setState({ counter: this.state.counter - 1 });
  };

  render() {
    return (
      <div>
        <div id="display">{this.state.counter}</div>
        <button onClick={this.increment}>+</button>
        <button onClick={this.decrement}>-</button>
      </div>
    );
  }
}

export default Counter;
