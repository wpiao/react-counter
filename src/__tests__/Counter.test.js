import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../components/Counter.jsx';

describe('<Counter /', () => {
  it('exists after my application has loaded', () => {
    const app = shallow(<Counter />);
    expect(app.find('div').exists()).toEqual(true);
    expect(app.find('button').exists()).toEqual(true);
  });

  it('should increment counter when click on increment button', () => {
    const app = mount(<Counter />);
    const increment = app.find('button').at(0);

    increment.simulate('click');
    expect(app.state('counter')).toEqual(1);
    expect(app.find('div').at(1).text()).toEqual('1');

    increment.simulate('click');
    expect(app.state('counter')).toEqual(2);
    expect(app.find('div').at(1).text()).toEqual('2');
  });

  it('sould decrement counter when click on decrement button', () => {
    const app = mount(<Counter />);
    const decrement = app.find('button').at(1);

    decrement.simulate('click');
    expect(app.state('counter')).toEqual(-1);
    expect(app.find('div').at(1).text()).toEqual('-1');

    decrement.simulate('click');
    expect(app.state('counter')).toEqual(-2);
    expect(app.find('div').at(1).text()).toEqual('-2');
  });

  it('should render correctly', () => {
    const counter = renderer.create(<Counter />).toJSON();
    expect(counter).toMatchSnapshot();
  });
});