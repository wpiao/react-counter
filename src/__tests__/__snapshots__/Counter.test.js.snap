// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter / should render correctly 1`] = `
<div>
  <div
    className="display"
  >
    0
  </div>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
